src/components/media-player/ha-media-player-browse.ts

```diff
@@ -286,19 +264,24 @@ export class HaMediaPlayerBrowse extends LitElement {
                               : "none",
                           })}
                         >
-                          ${child.can_expand && !child.thumbnail
+                          ${!child.thumbnail
                             ? html`
                                 <ha-svg-icon
                                   class="folder"
-                                  .path=${mdiFolder}
+                                  .path=${MediaClassBrowserSettings[
+                                    child.children_media_class ||
+                                      child.media_class
```

Comment:
We should only look at the `children_media_class` for directory/library
```suggestion
                                    child.children_media_class ||
                                      child.media_class
```